### PR TITLE
Dot language definition

### PR DESCRIPTION
**Pull request by SerekKiri**
This pull request adds dot language definition.

New language definition is placed in coalib/bearlib/languages/definitions/dot.py

Closes https://github.com/coala/coala/issues/5823

### CODE REVIEW

File: coalib/bearlib/languages/definitions/dot.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class dot:
+    aliases = 'DOT'
+    extensions = '.dot'
```
Comment (Dhiraj240) [inline coalib/bearlib/languages/definitions/dot.py]: ahh its ```.gv```

File: coalib/bearlib/languages/__init__.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 from .definitions.CSharp import CSharp
 from .definitions.CSS import CSS
 from .definitions.D import D
+from .definitions.Dot import dot
```
Comment (Dhiraj240) [inline coalib/bearlib/languages/__init__.py]: ```.definitions.Dot`` it should be similar to class name ```dot```

File: tests/TestUtilities.py [hunk 3]
```diff
@@ -40,6 +40,7 @@
     'CPP',
     'CSS',
     'D',
+    'Dot'
```
Comment (Dhiraj240) [inline tests/TestUtilities.py]: Here too ad it as class name.

**Comment by Dhiraj240** [comment]
@SerekKiri Better follow [this](https://en.wikipedia.org/wiki/DOT_(graph_description_language)

File: tests/TestUtilities.py [hunk 4]
```diff
@@ -40,6 +40,7 @@
     'CPP',
     'CSS',
     'D',
+    'DOT'
```
Comment (ayan-b) [inline tests/TestUtilities.py]: Missing comma.

**Comment by ayan-b** [comment]
In (these examples)[https://renenyffenegger.ch/notes/tools/Graphviz/examples/index], I see many keywords such as rank, parent etc.

File: coalib/bearlib/languages/definitions/DOT.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    comment_delimiters = '//',
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {"'": "'"}
+    indent_types = {'{': '}'}
+    encapsulators = {'(': ')', '[': ']'}
+    keywords = [
+        'node', 'edge', 'graph',
+        'digraph', 'subgraph', 'strict',
+        'shape', 'rank', 'parent',
+        'label',
```
Comment (ayan-b) [inline coalib/bearlib/languages/definitions/DOT.py]: `labelloc`, `port` these two are missing only. :smiley:

File: coalib/bearlib/languages/definitions/DOT.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class DOT:
+    aliases = 'DOT'
+    extensions = '.gv'
```
Comment (pareksha) [inline coalib/bearlib/languages/definitions/DOT.py]: Add a trailing comma in `aliases` and `extensions` as in other languages (to indicate it's a tuple).

Comment (SerekKiri) [reply coalib/bearlib/languages/definitions/DOT.py]: Done 👍

File: coalib/bearlib/languages/definitions/DOT.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class DOT:
+    aliases = 'DOT',
+    extensions = '.gv',
```
Comment (Man-Jain) [inline coalib/bearlib/languages/definitions/DOT.py]: There are two extensions .dot and .gv

File: coalib/bearlib/languages/definitions/DOT.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    comment_delimiters = '//',
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {"'": "'"}
+    indent_types = {'{': '}'}
+    encapsulators = {'(': ')', '[': ']'}
```
Comment (Man-Jain) [inline coalib/bearlib/languages/definitions/DOT.py]: Also add < and > delimiter here. Its given [here](https://www.graphviz.org/doc/info/lang.html)

File: coalib/bearlib/languages/definitions/DOT.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class DOT:
+    aliases = 'DOT',
+    extensions = '.gv',
+    comment_delimiters = '//',
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {"'": "'"}
```
Comment (Man-Jain) [inline coalib/bearlib/languages/definitions/DOT.py]: String delimiters are " " not ' '

File: coalib/bearlib/languages/definitions/DOT.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class DOT:
+    aliases = 'DOT',
+    extensions = '.gv',
+    comment_delimiters = '//',
```
Comment (Man-Jain) [inline coalib/bearlib/languages/definitions/DOT.py]: Lines starting with # are ignored. So you can also add that here.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5885/commits/a99704b9fffd7348bf4fa3d2045f446c11d16739

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**